pkg/cloud/azure/actuators/cluster/services.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure/actuators"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure/services"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure/services/certificates"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure/services/groups"
+)
+
+// Services are list of services required by cluster actuator, easy to create a fake
+type Services struct {
+	clusterName     string
+	groupsSvc       services.Service
+	certificatesSvc services.Service
+}
+
+// NewServices populates all the services based on input scope
+func NewServices(scope *actuators.Scope) (*Services, error) {
+	certificatesSvc, err := certificates.NewService(scope)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to create certificates service %q", scope.Cluster.Name)
+	}
+
+	groupsSvc, err := groups.NewService(scope)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to create groups service %q", scope.Cluster.Name)
+	}
+
+	return &Services{
+		clusterName:     scope.Cluster.Name,
+		groupsSvc:       groupsSvc,
+		certificatesSvc: certificatesSvc,
+	}, nil
+}
+
+// Reconcile reconciles all the services in pre determined order
+func (s *Services) Reconcile() error {
+	// Store cert material in spec.
+	if err := s.certificatesSvc.CreateOrUpdate(context.TODO()); err != nil {
```

Comment:
As mentioned earlier, we need to do something about passing the `context`

Reply:
Yep! completely agree, will address it in another PR, also its definitely in plan. Dont want to mix into this PR, since it would just bloat the whole change.